src/PostJob.ts

```diff
@@ -82,7 +104,38 @@ export default class PostJob {
         return;
     }
 
-    process() {
+    async process() {
+        if (!this.subGroup) {
+            const subGroupNameWithoutDivision = this.thread.name.replace(
+                `, ${getDivisionName(this.thread.name)},`,
+                ' '
+            );
+            //removes division name from subgroup to avoid repeating itself
+            try {
+                /*this.subGroup = await createNewSubgroup(
+                    this.groupId,
+                    subGroupNameWithoutDivision
+                );*/
+                this.setSubGroup(
+                    await createNewSubgroup(
+                        this.groupId,
+                        subGroupNameWithoutDivision
+                    )
+                );
+                console.log('New created group', this.subGroup);
+            } catch (error) {
+                if (error.status && error.status === 400) {
```

Comment:
handles 400 error, which triggers searching the group. what about other errors, for e.g. 500?

Reply:
How should other errors, such as a 500 be handled?

Reply:
if creating a subgroup did not go as planned, we should let the user know that something is wrong and not proceed with the processing of those files (since there will not be a subgroup to add them to, this.subgroup is undefined).